src/gh/diffCheck/diffCheck/df_vizualization.py

```diff
@@ -0,0 +1,293 @@
+"""
+
+import Rhino.Geometry as rg
+from System.Drawing import Color
+from diffCheck import df_vizualization
+
+
+class DFVizSettings:
+    """
+    This class compiles the settings for the vizualization into one object
+    """
+
+    def __init__(self, valueType, upper_threshold, lower_threshold, palette, legend_height, legend_width, legend_plane, histogram_scale_factor):
+
+        self.valueType = valueType
+
+        self.upper_threshold = upper_threshold
+        self.lower_threshold = lower_threshold
+        self.palette = df_vizualization.DFColorMap(palette)
+        self.legend_height = legend_height
+        self.legend_width = legend_width
+        self.legend_plane = legend_plane
+        self.histogram_scale_factor = histogram_scale_factor
+
+
+class DFColorMap:
+    """
+    This class defines different colormaps for visualization purposes
+    It allows selection of a colormap by name and initializes the corresponding color values.
+    """
+
+    def __init__(self, name):
+        self.names = name
+        if name == "Jet":
+            self.colors = [
+                Color.FromArgb(0, 0, 255),  # Blue
+                Color.FromArgb(0, 255, 255),  # Cyan
+                Color.FromArgb(0, 255, 0),  # Green
+                Color.FromArgb(255, 255, 0),  # Yellow
+                Color.FromArgb(255, 0, 0),  # Red
+            ]
+        elif name == "Rainbow":
+            self.colors = [
+                Color.FromArgb(127, 0, 255),
+                Color.FromArgb(0, 180, 235),
+                Color.FromArgb(128, 254, 179),
+                Color.FromArgb(255, 178, 96),
+                Color.FromArgb(255, 6, 3)
+            ]
+        elif name == "RdPu":
+            self.colors = [
+                Color.FromArgb(254, 246, 242),
+                Color.FromArgb(251, 196, 191),
+                Color.FromArgb(246, 103, 160),
+                Color.FromArgb(172, 1, 125),
+                Color.FromArgb(76, 0, 106)
+            ]
+        elif name == "Viridis":
+            self.colors = [
+                Color.FromArgb(68, 3, 87),
+                Color.FromArgb(58, 82, 139),
+                Color.FromArgb(32, 144, 140),
+                Color.FromArgb(94, 201, 97),
+                Color.FromArgb(248, 230, 33)
+            ]
+
+
+def interpolate_color(color1, color2, t):
+    """
+    Interpolate between two colors.
+    """
+
+    r = int(color1.R + (color2.R - color1.R) * t)
+    g = int(color1.G + (color2.G - color1.G) * t)
+    b = int(color1.B + (color2.B - color1.B) * t)
+
+    return Color.FromArgb(r, g, b)
+
+
+def value_to_color(value, min_value, max_value, palette):
+    """
+    Map a value to a color based on a colormap.
+    """
+
+    if value < min_value:
+        value = min_value
+    elif value > max_value:
+        value = max_value
+
+    colormap = palette.colors
+
+    # Normalize the value within the range
+    if min_value == max_value:
+        t = 0.5
+    else:
```

Comment:
Maybe we could add all these color utilities methods to the DFColorMap object? Even as static? Just to arrange a bit more

Reply:
- [X] Done